From 1ad5587b3d16fdfef88456bdfea2dae9ec840159 Mon Sep 17 00:00:00 2001
From: "Neal H. Walfield" <neal@pep.foundation>
Date: Fri, 6 Mar 2020 18:23:17 +0100
Subject: [PATCH] openpgp: Only build padding example if the required feature
 is on.

  - Padding requires zlib support.  If the "compression-deflate"
    feature is not set, then replace the example with a stub.

  - Fixes #448.
---
 examples/pad.rs | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/examples/pad.rs b/examples/pad.rs
index c1c341c8..af187a7f 100644
--- a/examples/pad.rs
+++ b/examples/pad.rs
@@ -12,10 +12,12 @@ use crate::openpgp::parse::Parse;
 use crate::openpgp::serialize::stream::{
     Message, LiteralWriter, Encryptor, Recipient,
 };
-use crate::openpgp::serialize::padding::*;
 use crate::openpgp::policy::StandardPolicy as P;
 
-fn main() {
+#[cfg(feature = "compression-deflate")]
+fn f() {
+    use crate::openpgp::serialize::padding::*;
+
     let p = &P::new();
     let args: Vec<String> = env::args().collect();
     if args.len() < 3 {
@@ -79,3 +81,8 @@ fn main() {
     // writer stack.
     literal_writer.finalize().unwrap();
 }
+
+fn main() {
+    #[cfg(feature = "compression-deflate")]
+    f()
+}
-- 
2.25.1

